sentry/src/main/java/io/sentry/SentryRuntimeEventProcessor.java

```diff
@@ -0,0 +1,48 @@
+package io.sentry;
+
+import io.sentry.protocol.SentryRuntime;
+import io.sentry.protocol.SentryTransaction;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+/** Attaches Java vendor and version to events and transactions. */
+final class SentryRuntimeEventProcessor implements EventProcessor {
+  private final @Nullable String javaVersion;
+  private final @Nullable String javaVendor;
+
+  public SentryRuntimeEventProcessor(
+      final @Nullable String javaVersion, final @Nullable String javaVendor) {
+    this.javaVersion = javaVersion;
+    this.javaVendor = javaVendor;
+  }
+
+  public SentryRuntimeEventProcessor() {
+    this(System.getProperty("java.version"), System.getProperty("java.vendor"));
+  }
+
+  @Override
+  public @Nullable SentryEvent process(
+      final @NotNull SentryEvent event, final @Nullable Object hint) {
+    return process(event);
+  }
+
+  @Override
+  public @Nullable SentryTransaction process(
+      final @NotNull SentryTransaction transaction, final @Nullable Object hint) {
+    return process(transaction);
+  }
+
+  private <T extends SentryBaseEvent> @Nullable T process(final @Nullable T event) {
```

Comment:
```suggestion
  private <T extends SentryBaseEvent> @Nullable T process(final @NotNull T event) {
```

the `event` cant be null, so it is `@NotNull`

Reply:
👍